libs/mips-binutils/Makefile

```diff
@@ -1,22 +1,66 @@
-.PHONY: all clean
+.PHONY: install_libelf uninstall_libelf
+.PHONY: install_mips-binutils uninstall_mips-binutils
+.PHONY: install uninstall
+.PHONY: clean
 
-all: lib/libgtest.a
 
-lib/libgtest.a: lib/gtest-all.o
-	ar -rv lib/libgtest.a lib/gtest-all.o
+#########################################################################
+#                             MIPS_BINUTILS
+#########################################################################
 
-lib/gtest-all.o: lib/src/gtest-all.cc
-	g++ -I lib/include -I lib -c $< -o $@
+# Variables for convinience:
+PREFIX=/usr/local/mips-binutils
+ALIAS_DIR=/usr/local/bin
 
-lib/src/gtest-all.cc: release-1.8.0.tar.gz
-	rm -rf lib
-	tar xfm release-1.8.0.tar.gz
-	mv googletest-release-1.8.0/googletest lib
-	rm -rf googletest-release-1.8.0
+AS= $(PREFIX)/bin/mipsel-unknown-linux-gnu-as
+LD= $(PREFIX)/bin/mipsel-unknown-linux-gnu-ld
+OB= $(PREFIX)/bin/mipsel-unknown-linux-gnu-objdump
+MIPS_UTILS= $(AS) $(LD) $(OB)
 
-release-1.8.0.tar.gz:
-	wget https://github.com/google/googletest/archive/release-1.8.0.tar.gz
+ALIAS_AS= $(ALIAS_DIR)/mips-as
+ALIAS_LD= $(ALIAS_DIR)/mips-ld
+ALIAS_OB= $(ALIAS_DIR)/mips-objdump
+ALIASES= $(ALIAS_AS) $(ALIAS_LD) $(ALIAS_OBJ)
 
+DIR= binutils
+ARCHIVE= binutils-2.28.tar.bz2
+URL= http://ftp.gnu.org/gnu/$(DIR)/$(ARCHIVE)
```

Comment:
I don't like the idea of automatized installation (both libelf and binutils) as nobody will maintain that code after you leave the project. However, manual installation is a good thing as an exercise for students, and in case of problem they may ask mentor and/or fix the Wiki page.